### PR TITLE
Add support for mysqldump --ignore-table switch

### PR DESCRIPTION
**Pull request by cosmix**
##### SUMMARY

Added support for `mysqldump`'s `--ignore-table` command line argument that provides for table exclusion in generated dumps. Includes integration testing of said feature.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
mysql_db

##### ANSIBLE VERSION

```
ansible 2.6
```

### CODE REVIEW

File: lib/ansible/modules/database/mysql/mysql_db.py
Comment (Akasurde) [inline lib/ansible/modules/database/mysql/mysql_db.py]: required is by-default 'false' so no need to specify explicitly.

**Review by Akasurde — COMMENTED** [review]
Rest LGTM

File: lib/ansible/modules/database/mysql/mysql_db.py
Comment (cosmix) [reply lib/ansible/modules/database/mysql/mysql_db.py]: Hello. Practically every single option already present in this module specifies it. I prefer explicit as it makes it absolutely clear, even to someone like me with practically no experience with this codebase. Given the above I don’t see any point in removing it from just this particular option.

Do you think I should remove it?

**Comment by daxlab** [comment]
>required is by-default 'false' so no need to specify explicitly.

@cosmix @Akasurde Explicit is better than implicit.

**Review by daxlab — APPROVED** [review]

**Comment by cosmix** [comment]
Hello @Akasurde, @daxlab could you please let me know how intend to proceed with this PR?

**Review by Akasurde — APPROVED** [review]
LGTM

**Review by daxlab — APPROVED** [review]
LGTM too.

**Comment by cosmix** [comment]
Hello @daxlab, @Akasurde — thanks for approving. Any idea when this would be merged?

**Comment by cosmix** [comment]
Hello @daxlab, @Akasurde, I hope you're doing well. Is there any additional issue preventing this PR from being merged?

**Comment by cosmix** [comment]
Hello @daxlab and @Akasurde I hope you're well — if there is something I can do to help get this PR merged please let me know. Does it require an additional review by the module maintainer? Someone else? Please advise.

**Comment by Akasurde** [comment]
@cosmix Could you please rebase this PR and add current version "2.6" in documentation ?

**Comment by cosmix** [comment]
@Akasurde done.

**Comment by cosmix** [comment]
@Akasurde Hello, this was rebased, as requested, more than a month ago. Could you please merge?

**Comment by Akasurde** [comment]
rebuild_merge

**Comment by Akasurde** [comment]
@cosmix Could you please address CI failures and ping me ?

**Comment by cosmix** [comment]
@akasurde okay, this is done, version updated once again. If there is nothing else, please merge.

**Comment by Akasurde** [comment]
@cosmix Thanks for the PR.

**Comment by cosmix** [comment]
@Akasurde My pleasure. Have a nice day.

**Pull request merged**
